pages/detail-ride/[slug].js

```diff
@@ -0,0 +1,86 @@
+import Heading from "../../components/Heading/Heading.styled";
+import SecondHeading from "../../components/SecondHeading/SecondHeading.styled";
+import { rides } from "../../lib/rides";
+import {
+  StyledCardForDetailPage,
+  StyledDescriptionTerm,
+  StyledDescriptionDetails,
+  StyledDateAndTime,
+  StyledSpan,
+  StyledPreCardForDetailPage,
+} from "../../components/Card/Card.styled";
+import Image from "next/image";
+import { useRouter } from "next/router";
+import { StyledLink } from "../../components/Link/Link.styled";
+
+export default function DetailPage() {
+  const router = useRouter();
+  const { slug } = router.query;
+  const extractedRide = rides.find((ride) => ride.slug === slug);
+
+  if (!extractedRide) {
+    return null;
+  }
+
+  const {
+    driver,
+    car,
+    date,
+    time,
+    departure,
+    meetingpoint,
+    destination,
+    seats,
+    price,
+    currency,
+  } = extractedRide;
+
+  return (
+    <main>
```

Comment:
The main part could be outsourced into components.

Reply:
do you have a idea how to do that? I thought the [slug] has to happen within the detail-ride page, that's why I did it all there. If I do two components, e.g- PreDetailCard and DetailCard, I am not sure how to implement the slug...

Reply:
One idea would be that the component you're building takes a prop. This could, for example, also be the slug. Based on the slug, the component can then work dynamically. I hope it was somewhat helpful. Best regards from me.

Reply:
You could do it as @louisweiser suggested. It would make the return statement of `pages/detail-ride/[slug].js` more readable if there is only a `DetailRide` component which takes the `slug` as a prop.

There is no need to do this now, it could also be part of a bigger refactoring.